### PR TITLE
Fix hierarchy under anatomical cluster (part 1)

### PR DESCRIPTION
**Pull request by gouttegd**
This PR is a first step towards fixing the hierarchy under [anatomical cluster](http://purl.obolibrary.org/obo/UBERON_0000477), as discussed in #2819 (especially [this comment](https://github.com/obophenotype/uberon/issues/2819#issuecomment-1490700354).

It clarifies the definition of _anatomical cluster_ to align it with the one used in FBbt, and moves out of the hierarchy all the terms that undoubtedly do not fit the definition.

Importantly, the re-classification of [hard palate](http://purl.obolibrary.org/obo/UBERON_0003216) as something else than an anatomical cluster fixes the unsatisfiability of [uvular muscle](http://purl.obolibrary.org/obo/UBERON_0010235) and related terms, because that muscle `has_muscle_insertion` to the hard palate, and the `has_muscle_insertion` relation applies to connected anatomical structures only (which a cluster is not).

### CODE REVIEW

File: src/ontology/uberon-edit.obo [hunk 1]
```diff
@@ -7238,27 +7238,15 @@ relationship: present_in_taxon NCBITaxon:4751 ! Fungi
...
 name: anatomical cluster
-def: "Anatomical group that has its parts adjacent to one another." [http://orcid.org/0000-0001-9114-8737]
+def: "Anatomical group whose component anatomical structures lie in close proximity to each other." [FBbt:00007277]
 subset: common_anatomy
 subset: upper_level
-xref: AAO:0010009
-xref: AEO:0000041
-xref: BILA:0000041
-xref: CARO:0000041
 xref: EHDAA2:0003041
```
Comment (bvarner-ebi) [inline src/ontology/uberon-edit.obo:7244]: Could not find this in EHDAA2 using OLS. Should it be removed as well?

File: src/ontology/uberon-edit.obo [hunk 2]
```diff
@@ -209360,7 +209348,7 @@ property_value: provenance_notes "This class was sourced from an external ontolo
 id: UBERON:3000809
 name: accessory articulation
 def: "Paired, additional vertebral articulations located medial to the intervertebral space, adjacent to the vertebral centrum." [AAO:Cannatella_1985]
-is_a: UBERON:0000477 ! anatomical cluster
+is_a: UBERON:0034921 ! multi organ part structure
```
Comment (bvarner-ebi) [inline src/ontology/uberon-edit.obo:209351]: Unsure if this refers to something that is "multi organ".

**Review by bvarner-ebi — APPROVED** [review]
Please see inline comments. 
Otherwise OK.

File: src/ontology/uberon-edit.obo (hunk 1 above)
Comment (gouttegd) [reply src/ontology/uberon-edit.obo:7244]: I can’t find it through Ontobee or BioPortal as well, so I agree, I’ll remove it.

File: src/ontology/uberon-edit.obo (hunk 2 above)
Comment (gouttegd) [reply src/ontology/uberon-edit.obo:209351]: Each vertebra is individually considered as a separate “organ” (a _vertebra_ is a _vertebral element_, which is a _skeletal element_, which is an _organ_), so I think it is consistent to say that an articulation between two vertebrae is a “multi organ part structure“ – it involves subparts of two different organs, which fits the definition of “multi organ part structure“.

But this makes me realise that this term can be classified even more precisely under _articulation_ (UBERON:0004905), which is a subclass of _multi organ part structure_. I’ll do that now.

**Review by bvarner-ebi — APPROVED** [review]

**Pull request merged**